tests/Integration/ScriptsTest.php

```diff
@@ -219,110 +222,52 @@ public function test_enqueue_js_api_no_secret(): void {
 	 * @uses \Parsely\Parsely::update_metadata_endpoint
 	 * @uses \Parsely\Scripts::register_scripts
 	 * @uses \Parsely\Scripts::script_loader_tag
-	 * @group insert-js
+	 * @group enqueue-js
 	 */
 	public function test_enqueue_js_api_with_secret(): void {
-		ob_start();
-		$post_array = $this->create_test_post_array();
-		$post       = $this->factory->post->create( $post_array );
-		$this->go_to( '/?p=' . $post );
 		self::$scripts->register_scripts();
-
 		self::set_options( array( 'api_secret' => 'hunter2' ) );
-
 		self::$scripts->enqueue_js_api();
-		$intermediate_output = ob_get_contents();
-		self::assertSame(
-			'',
-			$intermediate_output,
-			'Failed to confirm scripts were not printed by enqueue_js_api()'
-		);
 
-		self::assertTrue(
-			wp_script_is( 'wp-parsely-api', 'enqueued' ),
-			'Failed to confirm api script was enqueued when an API secret is set'
-		);
-
-		wp_print_scripts();
-		$output = ob_get_clean();
-
-
-		self::assertStringContainsString(
-			"<script type='text/javascript' id='wp-parsely-api-js-extra'>
-/* <![CDATA[ */
-var wpParsely = {\"apikey\":\"blog.parsely.com\"};
-/* ]]> */
-</script>",
-			$output,
-			'Failed to confirm "localized" data were embedded'
-		);
-
-		self::assertStringContainsString(
-			// phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
-			"<script type='text/javascript' src='" . esc_url( plugin_dir_url( PARSELY_FILE ) ) . 'build/init-api.js?ver=' . Parsely::VERSION . "' id='wp-parsely-api-js'></script>",
-			$output,
-			'Failed to confirm script tag was printed correctly'
+		// Confirm that API script is registered and enqueued.
+		$this->assert_script_statuses(
+			'wp-parsely-api',
+			array( 'registered', 'enqueued' )
 		);
 	}
 
 	/**
-	 * Make sure users can log in.
+	 * Make sure that disabling authenticated user tracking works.
 	 *
 	 * @covers \Parsely\Scripts::enqueue_js_tracker
 	 * @uses \Parsely\Parsely::api_key_is_missing
 	 * @uses \Parsely\Parsely::api_key_is_set
 	 * @uses \Parsely\Parsely::get_options
 	 * @uses \Parsely\Parsely::parsely_is_user_logged_in
-	 * @group insert-js
+	 * @group enqueue-js
 	 * @group settings
 	 */
-	public function test_user_logged_in(): void {
+	public function test_do_not_track_logged_in_users(): void {
 		TestCase::set_options( array( 'track_authenticated_users' => false ) );
 		$new_user = $this->create_test_user( 'bill_brasky' );
 		wp_set_current_user( $new_user );
 
-		ob_start();
+		self::$scripts->register_scripts();
 		self::$scripts->enqueue_js_tracker();
 
-		$intermediate_output = ob_get_contents();
-		self::assertSame(
-			'',
-			$intermediate_output,
-			'Failed to confirm scripts were not printed by enqueue_js_tracker()'
-		);
-
-		self::assertFalse(
-			wp_script_is( 'wp-parsely-api', 'registered' ),
-			'Failed to confirm API script was not registered'
-		);
-
-		self::assertFalse(
-			wp_script_is( 'wp-parsely-api', 'enqueued' ),
-			'Failed to confirm API script was not enqueued'
-		);
```

Comment:
This is no longer being tested - is that your intent?

Reply:
Yes, I left this out because the test is aimed to test what happens with the tracker when the user is logged in and the `track_authenticated_users` option is disabled. So I considered testing the API a bit irrelevant and removed it. Let me know if you'd prefer to have it though.

Reply:
The previous test was checking what happened to logged in users, compared to non-logged in users, so the presence of checking the status of the API before made sense.

If I asked you to prove (through tests) that the API script is _not_ affected for logged in users, we can no longer do that.

Reply:
I've included an assertion to verify that that API is unaffected for logged-in users. Feel free to resolve the conversation or comment if you think anything additional would be needed.